rpcs3/ds4_pad_handler.cpp

```diff
@@ -628,6 +630,8 @@ void ds4_thread::on_init(const std::shared_ptr<void>& _this)
 
 			devInfo = devInfo->next;
 		}
+
+		hid_free_enumeration(devInfo);
```

Comment:
Good catch on the missing free! But i think you'll need to free based on the initial pointer we get from hid_enumerate, as devInfo is null at this point